### PR TITLE
Added SetStart command to allow players to change their spawn location in the start zone

### PR DESCRIPTION
**Pull request by BoomShotKapow**
I apologize if this isn't the proper way of contributing to the timer, but I mostly just wanted to familiarize myself with SourcePawn and this was a feature I wanted to see in the timer.

### CODE REVIEW

**Comment by Nairdaa** [comment]
https://gist.github.com/Nickelony/b62e01f41b4c39d60ce7ba80f43cb79c


It's already somewhere in the resources on discord.

**Comment by Nairdaa** [comment]
https://discord.com/channels/389675819959844865/495188255604473856/722168119249993908

There it is. 

https://discord.com/channels/389675819959844865/495188255604473856/722168119249993908

**Comment by BoomShotKapow** [comment]
Oh well, it was fun to do and familiarize myself with SourcePawn, since I've never used it before. Thanks for letting me know about those plugins though,

**Comment by Nairdaa** [comment]
I mean don't get me wrong. You did a good job, but I don't think the devs want it merged. I'm for if you ask me.

**Comment by BoomShotKapow** [comment]
Yea, I wasn't sure how to contribute. It's why I did a draft pull request, but if any of it is useful, it's there for them.

File: addons/sourcemod/scripting/shavit-core.sp [hunk 1]
```diff
@@ -2068,6 +2113,19 @@ void StartTimer(int client, int track)
...
 }
 
+void SetStart(int client)
+{
+	if(!IsValidClient(client) || IsFakeClient(client))
+	{
+		return;
+	}
+	
+	Call_StartForward(gH_Forwards_SetStart);
```
Comment (kidfearless) [inline addons/sourcemod/scripting/shavit-core.sp:2123]: Why did you implement this as a forward? It'd make more sense to implement a native in shavit-zones that would handle that.

**Review by kidfearless — CHANGES_REQUESTED** [review]
> Yea, I wasn't sure how to contribute. It's why I did a draft pull request, but if any of it is useful, it's there for them.
contributions are always appreciated. A lot of people have used the set start plugin from the discord, and the changes to save start positions were also talked about a while back, but I didn't feel like they were necessary enough for me to add them.

onto the review of the pr, the implementation is pretty similar to the plugins mentions though there would need to be some changes before it would be accepted. The first is removal of the new database table. While it's performance is probably better, the purpose of client preferences is exactly for this kind of thing. I'm interested if anyone else has comments on whether it should be a new table or a client preference.

Other than a few minor tweaks the other big concern is creating a native in shavit-core to fire a forward that shavit-zones listens to in order to communicate between the two. I can think of one reason why you would do it this way, but I'm curious on what your actual reason is.

**Comment by kidfearless** [comment]
Also not a big deal for simpler features like this, but in the future ideally PR new features into very_good_yes so that if there is an issue with it that it can be sorted out before becoming official.

**Pull request closed without merging**